src/main/java/attendance/domain/Attendances.java

```diff
@@ -1,38 +1,89 @@
+    public void addAttendance(LocalDateTime dateTime) {
+        LocalDate date = dateTime.toLocalDate();
+        isDuplicateAttendance(date);
+        isValidateSchedule(date);
+        attendances.put(date, new Attendance(dateTime));
     }
 
-    public Optional<Attendance> findAttendance(LocalDate date) {
+    private void isDuplicateAttendance(LocalDate date) {
+        Optional<Attendance> attendance = getOptionalAttendance(date);
+        if (attendance.isPresent()) {
+            throw new AttendanceArgumentException(DUPLICATE_ATTENDANCE);
+        }
+    }
+
+    private void isValidateSchedule(LocalDate date) {
+        if (!systemDateTime.isWorkingDay(date)) {
+            var formatted = DateTimeFormatter.ofPattern(CANNOT_ATTENDANCE_WEEKEND_FORMAT).format(date);
+            throw new AttendanceArgumentException(formatted);
+        }
+    }
+
+    public Attendance getAttendance(LocalDate date) {
+        Optional<Attendance> attendance = getOptionalAttendance(date);
+        if (attendance.isEmpty() || attendance.get().isTruancy()) {
+            throw new AttendanceArgumentException(CANNOT_FIND_ATTENDANCE);
+        }
+        return attendance.get();
+    }
+
+    public Optional<Attendance> getOptionalAttendance(LocalDate date) {
         return Optional.ofNullable(attendances.get(date));
     }
 
-    public void remove(LocalDate date) {
-        attendances.remove(date);
+    public void modifyAttendance(LocalDateTime dateTime) {
+        LocalDate date = dateTime.toLocalDate();
+        isValidateSchedule(date);
+        attendances.put(date, new Attendance(dateTime));
+    }
+
+    public List<Attendance> getAttendancesWithTruancy() {
+        List<LocalDate> workingDates = systemDateTime.extractWorkingDays();
+        List<Attendance> attendanceHistory = new ArrayList<>();
+
+        for (LocalDate date : workingDates) {
+            var attendance = attendances.getOrDefault(date, Attendance.generateTruancy(date));
+            attendanceHistory.add(attendance);
+        }
+
+        return attendanceHistory;
     }
 
-    public void validateDuplicate(LocalDate date) {
-        if (attendances.containsKey(date)) {
-            throw new AttendanceArgumentException(DUPLICATE_DATE);
+    public Map<LocalDate, AttendanceStatus> updateStatics() {
+        List<LocalDate> workingDates = systemDateTime.extractWorkingDays();
+        Map<LocalDate, AttendanceStatus> attendanceStatus = new HashMap<>();
+        for (LocalDate date : workingDates) {
+            attendanceStatus.put(date, getOrAbsence(date));
         }
+        return attendanceStatus;
     }
 
-    public Map<LocalDate, Attendance> getAttendances() {
-        return attendances;
+    private AttendanceStatus getOrAbsence(LocalDate date) {
+        Optional<Attendance> optionalAttendance = getOptionalAttendance(date);
+        return optionalAttendance
+            .map(Attendance::status)
+            .orElse(AttendanceStatus.ABSENCE);
```

Comment:
```suggestion
        if (attendances.containsKey(date)) {
            return attendances.get(date).status();
        }
        return AttendanceStatus.ABSENCE;
```
어떤 코드가 더 와닿는 표현으로 보이시나요? `Optional`을 통한 stream처리는 가독성을 위한 것이 아니기때문에 왜 `Optional`이 사용되어야하는지에 대해서 고민해보시면 좋을 것 같네요.

Reply:
확실히 양 쪽 경우를 비교해 보니, if문을 사용하는 쪽이 복잡하지 않아 직관성이 더 좋네요.
Optional을 새로 배우며,  "보기 좋은 코드"를 너무 주관적으로 생각했던 것 같습니다.🤦‍♂️